### PR TITLE
SDN-5238: add hypershift networksegmentation presubmit to ovnk

### PR DESCRIPTION
**Pull request by jluhrsen**
optional: true

### CODE REVIEW

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-metal-ipi-ovn-dualstack-local-gateway-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.18-e2e-aws-ovn-single-node-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-networksegmentation

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-networksegmentation

**Comment by jluhrsen** [comment]
ok, this looks to be working now. will still need to add some cleanup and remove any debugging, but the [hosted cluster](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-networksegmentation/1841172974573457408/artifacts/e2e-aws-ovn-hypershift-conformance-networksegmentation/create-hostedcluster/build-log.txt) and [management cluster](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-networksegmentation/1841172974573457408/artifacts/e2e-aws-ovn-hypershift-conformance-networksegmentation/create-management-cluster/build-log.txt) now will have the NetworkSegmentation feature enabled

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-aws-encryption-provider-master-e2e-hypershift

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-aws-encryption-provider-release-4.19-e2e-hypershift

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-image-registry-release-4.19-e2e-hypershift

**Comment by jluhrsen** [comment]
the rehearsals:

[4.14-periodics-e2e-aws-ovn](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-aws-ovn/1843763224814555136) is not affected and [no extra features enabled in mgmt cluster](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-aws-ovn/1843763224814555136/artifacts/e2e-aws-ovn/create-management-cluster/build-log.txt)

[4.17-e2e-aws-ovn-hypershift-conformance-networksegmentation](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-pull-ci-openshift-ovn-kubernetes-release-4.17-e2e-aws-ovn-hypershift-conformance-networksegmentation/1843763224751640576) has NetworkSegmentation enabled in both the [mgmt cluster](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-pull-ci-openshift-ovn-kubernetes-release-4.17-e2e-aws-ovn-hypershift-conformance-networksegmentation/1843763224751640576/artifacts/e2e-aws-ovn-hypershift-conformance-networksegmentation/create-management-cluster/build-log.txt) as well as the [hosted cluster](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-pull-ci-openshift-ovn-kubernetes-release-4.17-e2e-aws-ovn-hypershift-conformance-networksegmentation/1843763224751640576/artifacts/e2e-aws-ovn-hypershift-conformance-networksegmentation/create-hostedcluster/build-log.txt) and passes

[master-e2e-aws-ovn-hypershift-conformance-networksegmentation](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-networksegmentation/1843763224680337408) has NetworkSegmentation enabled in both the [mgmt cluster](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-networksegmentation/1843763224680337408/artifacts/e2e-aws-ovn-hypershift-conformance-networksegmentation/create-management-cluster/build-log.txt) as well as the [hosted cluster](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-networksegmentation/1843763224680337408/artifacts/e2e-aws-ovn-hypershift-conformance-networksegmentation/create-hostedcluster/build-log.txt) and passes

[image-registry-master-e2e-hypershift](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-pull-ci-openshift-image-registry-master-e2e-hypershift/1843763224554508288) does not have [any custom features enabled](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-pull-ci-openshift-image-registry-master-e2e-hypershift/1843763224554508288/artifacts/e2e-hypershift/create-management-cluster/build-log.txt), as expected.

**Review by tssurya — APPROVED** [review]
/lgtm

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.18-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.19-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-image-registry-master-e2e-hypershift

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.18-e2e-aws-ovn-hypershift-conformance-techpreview

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-image-registry-master-e2e-hypershift

**Comment by tssurya** [comment]
@jluhrsen what's the status here? how are we doing..?

**Comment by jluhrsen** [comment]
> @jluhrsen what's the status here? how are we doing..?

there is a bug in/around hypershift that does not allow the full set of techpreview to be installed. We thought it was
fixed (by the hypershift team), but when I updated this PR to do it the way they asked it still fails. Slack thread here
says it will be on their list todo, but no news yet.

I can update this PR to do the semi-hack method of only installing NetworkSegmentation, since we know that works.
but, my initial feedback was to do it this current/more-clean way.

what's your opinion? wait for the bug to get fixed, or push to get the NeworkSegmentation only?

File: ci-operator/step-registry/hypershift/setup-nested-management-cluster/hypershift-setup-nested-management-cluster-chain.yaml [hunk 1]
```diff
@@ -44,23 +49,31 @@ chain:
...
+          --service-cidr=172.29.0.0/16 \
+          --cluster-cidr=10.124.0.0/14 \
+          --node-pool-replicas=${HYPERSHIFT_NODE_COUNT} \
+          --node-selector=hypershift.openshift.io/control-plane=true \
+          --release-image=${HYPERSHIFT_HC_RELEASE_IMAGE} \
+          --release-stream=${DEFAULT_RELEASE_STREAM} \
+          --namespace=${HYPERSHIFT_NAMESPACE}
+      )
+
+      if [[ $FEATURE_SET == "TechPreviewNoUpgrade" ]]; then
```
Comment (sjenning) [inline ci-operator/step-registry/hypershift/setup-nested-management-cluster/hypershift-setup-nested-management-cluster-chain.yaml:72]: The issue is here.  We don't want `TechPreviewNoUpgrade` on the prow job mgmt cluster, only on the test cluster created in the `hypershift-aws-create` chain.

**Review by sjenning — CHANGES_REQUESTED** [review]

**Comment by jluhrsen** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.18-e2e-aws-ovn-hypershift-conformance-techpreview

File: ci-operator/step-registry/hypershift/setup-nested-management-cluster/hypershift-setup-nested-management-cluster-chain.yaml (hunk 1 above)
Comment (jluhrsen) [reply ci-operator/step-registry/hypershift/setup-nested-management-cluster/hypershift-setup-nested-management-cluster-chain.yaml]: ok, I am trying it now. IIRC this was my first effort (the hacky way) and I found that the install would work, but tech preview features were still not enabled.

let's see how the rehearsal looks now.

File: ci-operator/step-registry/hypershift/aws/create/hypershift-aws-create-chain.yaml [hunk 2]
```diff
@@ -93,10 +97,15 @@ chain:
...
+      if [[ $FEATURE_SET == "TechPreviewNoUpgrade" ]]; then
+        COMMAND+=(--feature-set=TechPreviewNoUpgrade)
+      fi
+      
+      
       if [[ -n $EXTRA_ARGS ]]; then
         COMMAND+=(${EXTRA_ARGS})
       fi
-     
+
```
Comment (maiqueb) [inline ci-operator/step-registry/hypershift/aws/create/hypershift-aws-create-chain.yaml:108]:
```suggestion
     
```

**Review by maiqueb — APPROVED** [review]
Alright !

LGTM.

just caught a whitespace you've probably added by mistake. It's a nit you can ignore, since running all this CI again is costly. 

Only consider it if someone asks for additional code changes.

File: ci-operator/step-registry/hypershift/setup-nested-management-cluster/hypershift-setup-nested-management-cluster-chain.yaml (hunk 1 above)
Comment (jluhrsen) [reply ci-operator/step-registry/hypershift/setup-nested-management-cluster/hypershift-setup-nested-management-cluster-chain.yaml]: I remembered incorrectly. this does work. [here](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-pull-ci-openshift-ovn-kubernetes-release-4.18-e2e-aws-ovn-hypershift-conformance-techpreview/1849544643998388224/artifacts/e2e-aws-ovn-hypershift-conformance-techpreview/dump/artifacts/hostedcluster-903a01a56eae65bd1183/cluster-scoped-resources/config.openshift.io/featuregates.yaml) is the featuregates for the hosted cluster.

File: ci-operator/step-registry/hypershift/aws/create/hypershift-aws-create-chain.yaml (hunk 2 above)
Comment (jluhrsen) [reply ci-operator/step-registry/hypershift/aws/create/hypershift-aws-create-chain.yaml:108]: I ended up leaving it for readability's sake :)

**Comment by jluhrsen** [comment]
> Alright !
> 
> LGTM.
> 
> just caught a whitespace you've probably added by mistake. It's a nit you can ignore, since running all this CI again is costly.
> 
> Only consider it if someone asks for additional code changes.

Thanks for the review @maiqueb. I just double checked the rehearsal and looks like it's doing what we want. I pinged @sjenning  on slack (and now here) because I'll still need a hypershift project approve as well.

File: ci-operator/step-registry/hypershift/aws/create/hypershift-aws-create-chain.yaml [hunk 3]
```diff
@@ -145,6 +156,20 @@ chain:
...
+      set +e
+      timeout 60s bash -c '
+          while true; do
+            oc --kubeconfig="$SHARED_DIR"/nested_kubeconfig get clusteroperators
+            oc --kubeconfig="$SHARED_DIR"/nested_kubeconfig get nodes -A
+            oc --kubeconfig="$SHARED_DIR"/nested_kubeconfig get pods -A
+            oc --kubeconfig="$SHARED_DIR"/nested_kubeconfig get featuregate cluster -o yaml
+            sleep 15
+        done
+      '
```
[the remark this answers is not in the document]
Comment (jluhrsen) [reply ci-operator/step-registry/hypershift/aws/create/hypershift-aws-create-chain.yaml]: this PR is updated and hopefully as bare bones as possible now. nothing extra.

File: ci-operator/step-registry/hypershift/setup-nested-management-cluster/hypershift-setup-nested-management-cluster-chain.yaml [hunk 4]
```diff
@@ -61,6 +62,8 @@ chain:
         --release-image=${HYPERSHIFT_HC_RELEASE_IMAGE} \
         --release-stream=${DEFAULT_RELEASE_STREAM} \
         --namespace=${HYPERSHIFT_NAMESPACE}
+
+      "${COMMAND[@]}"
```
[the remark this answers is not in the document]
Comment (jluhrsen) [reply ci-operator/step-registry/hypershift/setup-nested-management-cluster/hypershift-setup-nested-management-cluster-chain.yaml]: completely reverted this file to upstream. no changes here any longer.

**Comment by sjenning** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.18-e2e-conformance

**Comment by sjenning** [comment]
/approve
/lgtm
@tssurya put a hold, not sure why

**Comment by jluhrsen** [comment]
/pj-rehearse ack

[here is a job](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-aws-ovn-conformance/1851006039407726592) that's not configured to use techpreview and the [featuresgate output](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-aws-ovn-conformance/1851006039407726592/artifacts/e2e-aws-ovn-conformance/dump/artifacts/hostedcluster-5033aff66bc1a2b4efc5/cluster-scoped-resources/config.openshift.io/featuregates.yaml) showing it's not

and [a job](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-pull-ci-openshift-ovn-kubernetes-release-4.17-e2e-aws-ovn-hypershift-conformance-techpreview/1851006039344812032) that is and [it's featuresgate output](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/56065/rehearse-56065-pull-ci-openshift-ovn-kubernetes-release-4.17-e2e-aws-ovn-hypershift-conformance-techpreview/1851006039344812032/artifacts/e2e-aws-ovn-hypershift-conformance-techpreview/dump/artifacts/hostedcluster-2017ed6b597c465f2c59/cluster-scoped-resources/config.openshift.io/featuregates.yaml)

**Review by tssurya — APPROVED** [review]
/lgtm

**Comment by tssurya** [comment]
> > /approve /lgtm @tssurya put a hold, not sure why
> 
> /hold cancel
> 
> she put the hold when I told her you wanted me to get rid of the jq yaml hack stuff. we should be good now

I had put hold because earlier we had the bug in hypershift due to which we were only running the UDN tests...but then Jamo told me hypershift solved the bug so then I put hold since it means we could do the full run...

**Pull request merged**